packages/utils/src/dal/mikro-orm/mikro-orm-repository.ts

```diff
@@ -717,7 +717,28 @@ export function mikroOrmBaseRepositoryFactory<T extends object = object>(
         persist: false,
       })
 
-      return { id: (created as any).id, ...data }
+      // Since our BigNumber implementation isn't implementing a custom type, we have no good way of knowing if a field is a BigNumber,
+      // so we simply rely on naming conventions. Try to make improvements to this either by changing the bigNumber definition or this logic
+      const bigNumberFields = Object.entries(created)
+        .filter(([key, _]) => key.startsWith("raw_"))
+        .flatMap(([key, value]) => {
+          const nonRawFieldKey = key.replace("raw_", "")
+          const nonRawField = created[nonRawFieldKey]
+          if (nonRawField === undefined) {
+            return []
+          }
+
+          return [
+            [key, value],
+            [nonRawFieldKey, nonRawField],
+          ]
+        })
+        .reduce((acc, [key, value]) => {
+          acc[key] = value
+          return acc
+        }, {})
+
+      return { id: (created as any).id, ...bigNumberFields, ...data }
```

Comment:
we could bring this closer to the mikroORM implementation that prevents us from hard coding `raw_`, here is a quick concept - [commit](https://github.com/medusajs/medusa/compare/fix/test?expand=1#diff-03588856044247b1b2d450be24ae9f199407d1dfd1089cdb17d89fcae6ca3f5eR718-R725)

MikroORM does a whole song and dance around diff comparators, which i think is very neat, but not necessary for our case for now. Here is a ref - [MikroORM code](https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/entity/EntityFactory.ts#L179-L180)

Reply:
Looks pretty neat to me – anything stopping us from moving forward with this?

Reply:
I think longer-term, we really need to revisit our implementation of BigNumber, and implement it using a custom type rather than this sort of wrapper around `Property`. But your suggestion is better than what I did @riqwan, so let me do that for now 👍

Reply:
@riqwan seems to be working well, nice suggestion, much nicer!